### ISSUE TITLE
Undefined reference when trying to compile prrte

### ISSUE THREAD

**Issue by JiakunYan** [openpmix/prrte#1689]
## Background information
I was trying to compile openpmix and prrte on my laptop.

### What version of the PMIx Reference RTE (PRRTE) are you using? (e.g., v2.0, v3.0, git master @ hash, etc.)
I am using prrte v3.0.0.

##### What version of PMIx are you using? (e.g., v4.2.0, git branch name and hash, etc.)
I am using openPMIx v4.2.3

### Please describe the system on which you are running

* Operating system/version: ubuntu 20.04
* Computer hardware: Intel(R) Core(TM) i7-8550U CPU @ 1.80GHz
* Network type: I don't think the network is involved here.

-----------------------------

## Details of the problem

I used the following commands to compile pmix.
```
cd pmix-4.2.3
mkdir build
cd build
../configure --prefix=/opt/pmix
make
make install
```

I used the following commands to compile prrte.
```
cd prrte-3.0.0
mkdir build
cd build
../configure --prefix=/opt/prrte --with-pmix=/opt/pmix
make
```

I encountered the following errors.

```shell
make[2]: Entering directory '/mnt/Windows/workspace/prrte-3.0.0/build/src/tools/prted'
  CC       prted.o
../../../../src/tools/prted/prted.c: In function ‘main’:
../../../../src/tools/prted/prted.c:234:13: warning: implicit declaration of function ‘pmix_argv_append_nosize’; did you mean ‘PMIx_Argv_append_nosize’? [-Wimplicit-function-declaration]
  234 |             pmix_argv_append_nosize(&prte_launch_environ, environ[i]);
      |             ^~~~~~~~~~~~~~~~~~~~~~~
      |             PMIx_Argv_append_nosize
../../../../src/tools/prted/prted.c:339:5: warning: implicit declaration of function ‘pmix_setenv’; did you mean ‘pmix_getenv’? [-Wimplicit-function-declaration]
  339 |     pmix_setenv("PMIX_MCA_compress_base_silence_warning", "1", true, &environ);
      |     ^~~~~~~~~~~
      |     pmix_getenv
../../../../src/tools/prted/prted.c:388:13: warning: implicit declaration of function ‘pmix_argv_free’; did you mean ‘pmix_argv_len’? [-Wimplicit-function-declaration]
  388 |             pmix_argv_free(cores);
      |             ^~~~~~~~~~~~~~
      |             pmix_argv_len
../../../../src/tools/prted/prted.c:556:16: warning: implicit declaration of function ‘pmix_argv_count’; did you mean ‘pmix_argv_len’? [-Wimplicit-function-declaration]
  556 |     naliases = pmix_argv_count(nonlocal);
      |                ^~~~~~~~~~~~~~~
      |                pmix_argv_len
../../../../src/tools/prted/prted.c:326:9: warning: ignoring return value of ‘pipe’, declared with attribute warn_unused_result [-Wunused-result]
  326 |         pipe(wait_pipe);
      |         ^~~~~~~~~~~~~~~
  CCLD     prted
/usr/bin/ld: prted.o: in function `main':
prted.c:(.text.startup+0x14e): undefined reference to `pmix_argv_append_nosize'
/usr/bin/ld: prted.c:(.text.startup+0x43b): undefined reference to `pmix_setenv'
/usr/bin/ld: prted.c:(.text.startup+0x535): undefined reference to `pmix_argv_free'
/usr/bin/ld: prted.c:(.text.startup+0xeb1): undefined reference to `pmix_argv_count'
/usr/bin/ld: prted.c:(.text.startup+0xf15): undefined reference to `pmix_argv_free'
/usr/bin/ld: prted.c:(.text.startup+0x1089): undefined reference to `pmix_argv_append_nosize'
/usr/bin/ld: prted.c:(.text.startup+0x10db): undefined reference to `pmix_argv_free'
/usr/bin/ld: prted.c:(.text.startup+0x16c1): undefined reference to `pmix_argv_append_nosize'
/usr/bin/ld: prted.c:(.text.startup+0x16da): undefined reference to `pmix_argv_append_nosize'
/usr/bin/ld: prted.c:(.text.startup+0x16ed): undefined reference to `pmix_argv_append_nosize'
/usr/bin/ld: prted.c:(.text.startup+0x1766): undefined reference to `pmix_argv_append_nosize'
/usr/bin/ld: prted.c:(.text.startup+0x1782): undefined reference to `pmix_argv_append_nosize'
/usr/bin/ld: prted.o:prted.c:(.text.startup+0x1794): more undefined references to `pmix_argv_append_nosize' follow
/usr/bin/ld: ../../../src/.libs/libprrte.so: undefined reference to `pmix_argv_append_unique_nosize'
/usr/bin/ld: ../../../src/.libs/libprrte.so: undefined reference to `pmix_argv_copy'
/usr/bin/ld: ../../../src/.libs/libprrte.so: undefined reference to `pmix_argv_split_with_empty'
/usr/bin/ld: ../../../src/.libs/libprrte.so: undefined reference to `pmix_argv_split'
/usr/bin/ld: ../../../src/.libs/libprrte.so: undefined reference to `pmix_argv_join'
collect2: error: ld returned 1 exit status
```

**Comment by rhc54** [comment]
You probably need to stick with PMIx v4.2.2 or maybe even something a little earlier for PRRTE v3.0 as the standard moved those functions to a different name. Alternatively, you can use the head of PRRTE's v3.0 branch as it is tracking the changes and should be released as v3.0.1 soon.

**Comment by JiakunYan** [comment]
PMIx v4.2.2 works. Thank you!
